server/src/main/java/org/elasticsearch/action/admin/cluster/stats/CCSTelemetrySnapshot.java

```diff
@@ -0,0 +1,398 @@
+        out.writeVLong(successCount);
+        out.writeMap(failureReasons, StreamOutput::writeLong);
+        took.writeTo(out);
+        tookMrtTrue.writeTo(out);
+        tookMrtFalse.writeTo(out);
+        out.writeVLong(remotesPerSearchMax);
+        out.writeDouble(remotesPerSearchAvg);
+        out.writeVLong(skippedRemotes);
+        out.writeMap(featureCounts, StreamOutput::writeLong);
+        out.writeMap(clientCounts, StreamOutput::writeLong);
+        out.writeMap(byRemoteCluster, StreamOutput::writeWriteable);
+    }
+
+    public long getTotalCount() {
+        return totalCount;
+    }
+
+    public long getSuccessCount() {
+        return successCount;
+    }
+
+    public Map<String, Long> getFailureReasons() {
+        return Collections.unmodifiableMap(failureReasons);
+    }
+
+    public LongMetricValue getTook() {
+        return took;
+    }
+
+    public LongMetricValue getTookMrtTrue() {
+        return tookMrtTrue;
+    }
+
+    public LongMetricValue getTookMrtFalse() {
+        return tookMrtFalse;
+    }
+
+    public long getRemotesPerSearchMax() {
+        return remotesPerSearchMax;
+    }
+
+    public double getRemotesPerSearchAvg() {
+        return remotesPerSearchAvg;
+    }
+
+    public long getSearchCountWithSkippedRemotes() {
+        return skippedRemotes;
+    }
+
+    public Map<String, Long> getFeatureCounts() {
+        return Collections.unmodifiableMap(featureCounts);
+    }
+
+    public Map<String, Long> getClientCounts() {
+        return Collections.unmodifiableMap(clientCounts);
+    }
+
+    public Map<String, PerClusterCCSTelemetry> getByRemoteCluster() {
+        return Collections.unmodifiableMap(byRemoteCluster);
+    }
+
+    public static class PerClusterCCSTelemetry implements Writeable, ToXContentFragment {
+        private long count;
+        private long skippedCount;
+        private final LongMetricValue took;
+
+        public PerClusterCCSTelemetry() {
+            took = new LongMetricValue();
+        }
+
+        public PerClusterCCSTelemetry(long count, long skippedCount, LongMetricValue took) {
+            this.took = took;
+            this.skippedCount = skippedCount;
+            this.count = count;
+        }
+
+        public PerClusterCCSTelemetry(PerClusterCCSTelemetry other) {
+            this.count = other.count;
+            this.skippedCount = other.skippedCount;
+            this.took = new LongMetricValue(other.took);
+        }
+
+        public PerClusterCCSTelemetry(StreamInput in) throws IOException {
+            this.count = in.readVLong();
+            this.skippedCount = in.readVLong();
+            this.took = LongMetricValue.fromStream(in);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVLong(count);
+            out.writeVLong(skippedCount);
+            took.writeTo(out);
+        }
+
+        public PerClusterCCSTelemetry add(PerClusterCCSTelemetry v) {
```

Comment:
How about renaming 'v' to something more desriptive, like 'valueToAdd' or 'telemetryData'?